spec/worker_spec.rb

```diff
@@ -40,119 +40,106 @@ def maybe_cleanup_file(file_path)
                                :timeout => timeout
       end
 
-      # Eagerly initializing work_time and the like so that
-      # we don't introduce deadlocks inside of our subject. Believe
-      # this is related to RSpec 3.3 thread-safety changes: subject/let
-      # blocks are now sychronized across multiple threads. It's possible that the
-      # interaction between Celluloid (threading library for Sidekiq) and RSpec
-      # is causing the deadlock, but it's not exactly clear why.
-      # See for possible explanation: https://github.com/rspec/rspec-core/issues/2064
-      let!(:expiration) { work_time * 4 }
-      let!(:work_time) { 0.1 }
-
-      subject do
-        threads = []
-        threads << Thread.new { FakeWorker.new.perform(work_time) }
-        threads << Thread.new do
-          # TODO(jaredjenkins): figure out why this is causing a deadlock
-          # with out the eager loading the RSpec lets.
-          sleep wait_time
-          FakeWorker.new.perform(work_time)
-        end
-        threads.select(&:alive?).each(&:join)
+      let(:expiration) { work_time * 4 }
+      let(:work_time) { 0.1 }
+
+      def run_synchronized_workers
```

Comment:
Isn't this a little risky since it can be triggered multiple times now? i.e. not memoized like `subject`